.github/workflows/build.yml

```diff
@@ -146,8 +146,73 @@ jobs:
     - name: Image digest
       run: echo ${{ steps.build.outputs.digest }}
 
+  build-windows:
+    needs: version
+    runs-on: windows-latest
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      fail-fast: false
+      matrix:
+        base-image:
+          - mcr.microsoft.com/windows/servercore:ltsc2022
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Set image variables
+        id: image-variables
+        run: |
+          $baseImage = "${{ matrix.base-image }}"
+          $tag = if ($baseImage -like "*nanoserver*") { "nanoserver" } else { "servercore" }
+          echo "tag=$tag" >> $env:GITHUB_OUTPUT
+        shell: pwsh
+
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push Docker image
```

Comment:
i'm assuming docker build and push action doesn't work on windows?

Reply:
buildx was not compatible with windows when I tried